Can't build rock on cygwin

I'm lazy, so here's logs from IRC:

```
<minirop> I'm trying to compile rock under cygwin and I get : In function `lang_Buffer__Buffer_toLDouble_impl': /home/amos/ooc/rock/sdk/lang/Buffer.ooc:590: undefined reference to `_strtold'
<scott_> hmm, it appears cygwin doesn't handle long doubles
<scott_> I guess we could add a version block for __CYGWIN__ that would make strtold(a,b) just do strtod(a,b) as LDouble
```

--- a/sdk/lang/Character.ooc
+++ b/sdk/lang/Character.ooc
@@ -14,7 +14,15 @@ strtoll: extern func (Char*, Pointer, Int) -> LLong
 strtoul: extern func (Char*, Pointer, Int) -> ULong
 strtof:  extern func (Char*, Pointer)      -> Float
 strtod:  extern func (Char*, Pointer)      -> Double
-strtold: extern func (Char*, Pointer)      -> LDouble
+// Cygwin doesn't handle long doubles (and doesn't have strtold defined)
+version(!__CYGWIN__) {
+    strtold: extern func (Char*, Pointer)      -> LDouble
+}
+version(__CYGWIN__) {
+    strtold: func (str: Char*, p: Pointer) -> LDouble {
+        strtod(str, p) as LDouble
+    }
+}
 
 /**
  * Character type

